### ISSUE TITLE
Move `a/b.p.detailedNDens` to internal plugin

### ISSUE THREAD

**Issue by keckler** [terrapower/armi#1702]
The parameter `detailedNDens` is defined for both blocks and assemblies in the framework. Here are both of those definitions:
https://github.com/terrapower/armi/blob/1e356e7ee1756c4bde05eea1a9b9a8398e72c4f1/armi/reactor/assemblyParameters.py#L134-L147
https://github.com/terrapower/armi/blob/1e356e7ee1756c4bde05eea1a9b9a8398e72c4f1/armi/reactor/blockParameters.py#L51-L65

There are a couple minor issues with these parameters being defined here:
1. They only make sense within the context of the internal TP depletion solver, and are thus confusing to anyone that doesn't have access to that internal solver
2. These parameters are only set if a specific setting is turned on (`burnxDetailedDepletion`), and that setting is internal to TP's ARMI app
3. This parameter is seemingly duplicate with the other number density block parameters (`nFe56`, `nU238`, etc.) and is thus confusing

In order to disambiguate the situation, it would be great if we could add context to the parameter's description that, for instance, it is only set when `burnxDetailedDepletion` is turned on. In order to do that, the setting would have to be made internal.

I think there would be value in bringing this hyper-specific setting internal. The setting definition says as much in a comment, so I don't anticipate this would break anything.

**Comment by john-science** [comment]
The question in ARMI is always "Is this idea generally useful to reactor modeling". Now, maybe some ideas are specific to LWRs or SFRs or hex-assemblies, but those are still worth including in ARMI for a good chunk of the target audience. In that light, this `Parameter` feels pretty borderline to me;  _maybe_ someone else would want this `Parameter`. But, this was clearly designed with a very specific implementation of our internal code in mind.

> I agree with the proposal; it feels cleaner to put this `Parameter` with the depletion code that it was designed for.

**Comment by albeanth** [comment]
> In that light, this Parameter feels pretty borderline to me; maybe someone else would want this Parameter.

I've always thought that ARMI had too many parameters that aren't actually tied to anything in ARMI. We just assume that people will want to use them (which I think is a questionable assumption, personally). I imagine that most people, if they're taking the time and effort to write their own ARMI Application, will want to write their own parameters and not rely on the defaults in ARMI. 

I think it would be cleaner (and easier to maintain by not having to ask permission or surprisingly wrecking someone's App) if we only left the most obvious ones in ARMI (e.g., flux, keff, pitch, other geometric info, etc) or ones that are actually used in ARMI. These could serve as examples for others to develop their own.

Thoughts?

**Comment by keckler** [comment]
> I've always thought that ARMI had too many parameters that aren't actually tied to anything in ARMI. We just assume that people will want to use them (which I think is a questionable assumption, personally). I imagine that most people, if they're taking the time and effort to write their own ARMI Application, will want to write their own parameters and not rely on the defaults in ARMI.

I very much agree with this sentiment. It is not difficult for the creator of an App to define their own parameters. I think that stuffing the open-source ARMI with maybe-useful parameters is only to our detriment.

**Comment by john-science** [comment]
> I think it would be cleaner iif we only left the most obvious ones in ARMI

Agreed. So far, we have removed ~150 Parameters from ARMI.  But these _are_ breaking changes, and some of those parameters _are_ used downstream, so they have to be moved carefully rather than just deleted.  I _was_ able to just straight delete 100-120 parameters, but that did take weeks of discussion to do.

If there is support, I would be happy to do more. Perhaps there are bulk groupings of parameters we could consider all at one time. Say, if there was a particular TH or neutronics application that wanted to take responsibility for a bunch of ARMI parameters and move them out of ARMI... that could be really helpful.  :)

**Comment by albeanth** [comment]
> Say, if there was a particular TH or neutronics application that wanted to take responsibility for a bunch of ARMI parameters and move them out of ARMI... that could be really helpful. :)

I'll start an internal discussion!

**Comment by john-science** [comment]
Okay, after some discussion, we've decided to just move this parameter out of ARMI.

**Comment by john-science** [comment]
Also, the `detailedNDens` param and the `detailedNucKeys` parameter are completely tied together, and can not be separated.

**Comment by john-science** [comment]
Drat.

This parameter is actually used directly in ARMI:

https://github.com/terrapower/armi/blob/9ea428bbbc196e80729e6153b2dc7e249d744aad/armi/reactor/blocks.py#L727-L745

It is called in one place:

https://github.com/terrapower/armi/blob/9ea428bbbc196e80729e6153b2dc7e249d744aad/armi/reactor/blocks.py#L685-L706

So, this probably isn't a simple copy/paste any more.

Let me look at this.
